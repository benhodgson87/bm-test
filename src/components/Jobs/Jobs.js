import React from 'react'
import PropTypes from 'prop-types'
import { Row } from 'react-flexbox-grid'
import { Alert } from 'react-bootstrap'

import JobItem from '../JobItem'

const Jobs = ({ pending, items }) => (
  <div className="jobs-list">
    {pending && <Row className="jobs-list-isLoading">Loading</Row>}
    {!pending &&
      items.length < 1 && (
        <Row className="jobs-list-isEmpty">
          <Alert bsStyle="warning">No items to display</Alert>
        </Row>
      )}
    {!pending &&
      items.length > 0 &&
      items.map(item => (
        <JobItem
          key={item.id}
          id={item.id}
          customer={item.customer}
          dimensions={item.job.dimensions}
        />
      ))}
  </div>
)

Jobs.propTypes = {
  pending: PropTypes.bool.isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default Jobs
