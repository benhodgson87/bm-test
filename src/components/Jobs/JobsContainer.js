import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getJobListing } from '../../store/jobs/thunks'

import Jobs from './Jobs'

export class JobsContainer extends Component {
  static propTypes = {
    isPending: PropTypes.bool,
    jobListings: PropTypes.arrayOf(PropTypes.object),
    getCurrentJobs: PropTypes.func.isRequired,
  }

  static defaultProps = {
    jobListings: [],
    isPending: false,
  }

  componentWillMount() {
    const { getCurrentJobs } = this.props
    getCurrentJobs()
  }

  render() {
    const { jobListings, isPending } = this.props
    return <Jobs items={jobListings} pending={isPending} />
  }
}

export const mapStateToProps = ({ jobs }) => ({
  jobListings: jobs.data,
  isPending: jobs.pending,
})

export const mapDispatchToProps = dispatch => ({
  getCurrentJobs: () => dispatch(getJobListing()),
})

export default connect(mapStateToProps, mapDispatchToProps)(JobsContainer)
