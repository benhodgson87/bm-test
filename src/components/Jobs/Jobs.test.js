import React from 'react'
import { shallow, mount } from 'enzyme'

import Jobs from './Jobs'

const baseProps = {
  pending: false,
  items: [
    {
      id: 123,
      customer: {
        title: 'Mr',
        first_name: 'Wayne',
        last_name: 'World',
      },
      job: {
        dimensions: 234,
      },
    },
  ],
}

const renderProps = props => Object.assign({}, baseProps, props)

describe('Jobs presentational component', () => {
  test('renders as expected', () => {
    const wrapper = shallow(<Jobs {...renderProps()} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('shows the loading text if pending', () => {
    const props = { pending: true }
    const wrapper = mount(<Jobs {...renderProps(props)} />)
    expect(wrapper.text()).toContain('Loading')
  })

  test('shows an empty message if jobs list length is zero', () => {
    const props = { items: [] }
    const wrapper = mount(<Jobs {...renderProps(props)} />)
    expect(wrapper.text()).toContain('No items to display')
  })
})
