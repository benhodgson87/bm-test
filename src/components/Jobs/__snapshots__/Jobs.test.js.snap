// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs presentational component renders as expected 1`] = `
<div
  className="jobs-list"
>
  <JobItem
    customer={
      Object {
        "first_name": "Wayne",
        "last_name": "World",
        "title": "Mr",
      }
    }
    dimensions={234}
    id={123}
    key="123"
  />
</div>
`;
